### PR TITLE
chore(observability)!: Only refer to components by id rather than name

### LINKED ISSUE #8408: Settle on calling the component identifiers `id` rather than `name`
For the following config:

```toml
[transforms.parse_nginx]
type = "remap"
# ...
```

We refer to the `parse_nginx` part as both the name (as `component_name` in metric tags and log fields) but as `id` in other places like the docs. After some discussion, we've decided to settle on `id` as:

* This leaves space for `name` to be a human readable name
* `id` gives more of the connotation that it should not change

This will be a breaking change that we should highlight in the release it goes out in.

We should update `component_name` to `component_id` and any references to the identifier to `id` in the docs.

See https://github.com/timberio/vector/pull/8337 for a start.

### PR DESCRIPTION
**Pull request by jszwedko**
This PR makes a pass to drop all references to "component name"
preferring "component id" as we believe id is closer to the intent of
the field than name is. Name we expect we may re-add later and have it
be a human readable name.

I'm sure I missed places in the docs and internal variable and type
names, but I believe this PR covers the breaking interface changes:

* Tagging metrics with `component_id` rather than `component_name`
* Removing the `name` field from `Component` in the GraphQL API,
  replacing it with `componentId`

See upgrade the added 0.16.0 upgrade guide.

An alternative approach to this would be to, instead of removing `name`
right away, add `id`. For example, have an additional metric tag of
`component_id` but leave `component_name` too. We could then remove
`component_name` in a subsequent release. I personally feel the impact
should be small enough that users can make the change quickly if needed.
I expect it should just be to update metrics queries in their downstream
systems.

Closes #8408 

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

**Comment by jszwedko** [comment]
@leebenson do you mind taking a peek at the GraphQL, `top`, and `tap` changes?

File: docs/content/en/guides/advanced/custom-aggregations-with-lua.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ Using the knowledge from the previous section, it is possible to write down the
 [transforms.aggregator]
 type = "lua"
 version = "2"
-inputs = [] # add names of the input components here
+inputs = [] # add ids of the input components here
```
[the remark this answers is not in the document]
Comment (spencergilbert) [reply docs/content/en/guides/advanced/custom-aggregations-with-lua.md:83]: (same for the other two 'ids' in this file

File: docs/content/en/guides/getting-started/getting-started.md [hunk 2]
```diff
@@ -50,15 +50,15 @@ Let's create a configuration file:
   encoding.codec = "text"
 ```
 
-Each component has a unique name and is prefixed with the type of component, for example, `sources` for a
+Each component has a unique id and is prefixed with the type of component, for example, `sources` for a
```
Comment (spencergilbert) [inline docs/content/en/guides/getting-started/getting-started.md:53]:
```suggestion
Each component has a unique ID and is prefixed with the type of component, for example, `sources` for a
```

File: docs/content/en/highlights/2021-01-20-wildcard-identifiers.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
 date: "2021-01-20"
-title: "Wildcards are now supported in component names"
+title: "Wildcards are now supported in component ids"
```
Comment (spencergilbert) [inline docs/content/en/highlights/2021-01-20-wildcard-identifiers.md:3]:
```suggestion
title: "Wildcards are now supported in component IDs"
```

File: docs/content/en/highlights/2021-01-20-wildcard-identifiers.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ badges:
   domains: ["topology"]
 ---
 
-[PR 6170][pr_6170] introduced wildcards when referencing component names in the `inputs` option. This allows you to build
+[PR 6170][pr_6170] introduced wildcards when referencing component ids in the `inputs` option. This allows you to build
```
Comment (spencergilbert) [inline docs/content/en/highlights/2021-01-20-wildcard-identifiers.md:15]:
```suggestion
[PR 6170][pr_6170] introduced wildcards when referencing component IDs in the `inputs` option. This allows you to build
```

**Review by spencergilbert — COMMENTED** [review]
There's a number of places (outside of code) where we switch between "ID" and "id" we should stick to one.

We could also continue to tag metrics with `component_name` to reduce the breakage between versions, and remove it later down the line

File: docs/content/en/blog/graphql-api.md [hunk 5]
```diff
@@ -64,7 +64,7 @@ events, bytes, and errors processed per component:
 query {
   components {
     __typename
-    name
+    componentI
```
[the remark this answers is not in the document]
Comment (jszwedko) [reply docs/content/en/blog/graphql-api.md:67]: Doh, thanks :)

File: docs/content/en/guides/advanced/custom-aggregations-with-lua.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jszwedko) [reply docs/content/en/guides/advanced/custom-aggregations-with-lua.md:83]: Yeah, it seems reasonable to capitalize them in the docs.

**Comment by jszwedko** [comment]
Yeah, good call @blt (and others). I've updated this PR to leave `component_name` on metrics for now; just marking it as deprecated.

File: src/topology/running.rs [hunk 6]
```diff
@@ -463,169 +463,175 @@ impl RunningTopology {
...
+    fn spawn_sink(&mut self, id: &str, new_pieces: &mut builder::Pieces) {
+        let task = new_pieces.tasks.remove(id).unwrap();
         let span = error_span!(
             "sink",
+            component_id = %task.id(),
             component_kind = "sink",
-            component_name = %task.name(),
             component_type = %task.typetag(),
+            // maintained for compatibility
+            component_name = %component_id,
```
Comment (spencergilbert) [inline src/topology/running.rs:507]: Looks like this should be:
```suggestion
            component_name = %task.id,
```

File: src/topology/running.rs [hunk 7]
```diff
@@ -463,169 +463,175 @@ impl RunningTopology {
...
+    fn spawn_transform(&mut self, id: &str, new_pieces: &mut builder::Pieces) {
+        let task = new_pieces.tasks.remove(id).unwrap();
         let span = error_span!(
             "transform",
             component_kind = "transform",
-            component_name = %task.name(),
+            component_id = %task.id(),
             component_type = %task.typetag(),
+            // maintained for compatibility
+            component_name = %component_id,
```
Comment (spencergilbert) [inline src/topology/running.rs:524]:
```suggestion
            component_name = %task.id,
```

File: src/topology/running.rs [hunk 8]
```diff
@@ -463,169 +463,175 @@ impl RunningTopology {
...
+    fn spawn_source(&mut self, id: &str, new_pieces: &mut builder::Pieces) {
+        let task = new_pieces.tasks.remove(id).unwrap();
         let span = error_span!(
             "source",
             component_kind = "source",
-            component_name = %task.name(),
+            component_id = %task.id(),
             component_type = %task.typetag(),
+            // maintained for compatibility
+            component_name = %component_id,
```
Comment (spencergilbert) [inline src/topology/running.rs:541]:
```suggestion
            component_name = %task.id,
```

**Review by spencergilbert — APPROVED** [review]

File: src/topology/running.rs (hunk 6 above)
Comment (jszwedko) [reply src/topology/running.rs:507]: Yes indeed, thanks!

Comment (jszwedko) [reply src/topology/running.rs]: 6345ff8

**Pull request merged**